src/Components/PowerBIWidget/Internal/PowerBIWidgetBuilder/PowerBIWidgetBuilderAdapter.ts

```diff
@@ -0,0 +1,71 @@
+import { IAuthService } from '../../../../Models/Constants';
+import { IPowerBIWidgetBuilderAdapter } from './PowerBIWidgetBuilder.types';
+import { service, factories, Report } from 'powerbi-client';
+import { createGUID } from '../../../../Models/Services/Utils';
+import { TokenType } from 'powerbi-models';
+
+const powerBIService = new service.Service(
+    factories.hpmFactory,
+    factories.wpmpFactory,
+    factories.routerFactory
+);
+export default class PowerBIWidgetBuilderAdapter
+    implements IPowerBIWidgetBuilderAdapter {
+    authService: IAuthService;
+    element: HTMLDivElement;
+    report: Report;
+    constructor(authService: IAuthService) {
+        this.authService = authService;
+        // Power BI SDK relies on being loaded into an HTML elevement
+        const guid = createGUID();
+        this.element = document.createElement('div');
+        this.element.id = guid;
```

Comment:
The Power BI SDK needs to "load" a report into a DOM element before you can make calls to query things. So I'm making a hidden within the adapter to encapsulate all these calls so that from the outside it "feels" more traditional REST-y service call.